### ISSUE TITLE
Error 454

### ISSUE THREAD

**Issue by lakenen** [ryancole/topdf#2]
I moved my workspace to a different VM, this one running Ubuntu 11.10, and now I get this error `{ step: 2, error: 454 }` every time I try to run the test. Any idea why that would happen? I looked it up in `scerr.h` and I believe the error is:

``` cpp
#define SCCERR_NOGRAPHICEXPORTFILTER  0x1C6 /* No export filter available to translate to selected graphic type. */
```

**Comment by ryancole** [comment]
Did you copy the entire contents of the SDK's `redist` directory into `/usr/local/lib/pdfexport`? Also, what's the file type of the input file? The error seems to suggest that you're missing the export filters from the redist directory, but I've seen this error in my other module and I had the redist files copied, so it could also be something else.

That's the correct error that you mentioned, though. I will boot up an Ubuntu 11.10 VM and see if I can debug this.

**Comment by lakenen** [comment]
Yep, I made sure I copied everything in `redist` over to `/usr/local/lib/pdfexport`. I am just using the `foo.xlsx` that you included in the repo.

**Comment by ryancole** [comment]
Alright. The Ubuntu ISO is downloading. Hopefully it's not something too obscure, because support from Oracle on this lib is not the easiest thing to come across. :(

**Comment by ryancole** [comment]
Well, it's running for me on Ubuntu 11.10. I wonder what's different between our two setups? Do you have libstdc++5 installed? You would probably get an error if not, so that's probably not it. Perhaps a permissions issue on the SDK lib directory? I started with a bare Ubuntu 11.10 install and just installed: build-essential, the fonts, libstdc++5 and node 0.9.3 from source.

**Comment by lakenen** [comment]
Hmm might be my node version. I'll try updating to 0.9.3.

**Comment by ryancole** [comment]
Unless you're on a much older version, I doubt it'll be the node version. The Oracle PDF Export version that I'm using is `8.4.0`. Below is a list of the contents of my `/usr/local/lib/pdfexport` directory:

```
ryan@ubuntu:~$ ls /usr/local/lib/pdfexport/
adinit.dat    imcd82.flt  imrnd2.flt             libsc_ca.so    libvs_acs.so   libvs_en4.so   libvs_html.so  libvs_lzh.so   libvs_mwkd.so  libvs_pic.so   libvs_qt.so    libvs_snap.so  libvs_txt.so   libvs_word.so  libvs_xml.so
cmmap000.bin  imcdr2.flt  iphgw2.flt             libsc_ch.so    libvs_ami.so   libvs_ens.so   libvs_hwp2.so  libvs_m11.so   libvs_mwks.so  libvs_pict.so  libvs_rar.so   libvs_so6.so   libvs_vcrd.so  libvs_work.so  libvs_xps.so
ibfpx2.flt    imcm52.flt  isunx2.flt             libsc_da.so    libvs_arc.so   libvs_enw.so   libvs_hwp.so   libvs_manu.so  libvs_mwp2.so  libvs_png.so   libvs_ras.so   libvs_soc6.so  libvs_viso.so  libvs_wp5.so   libvs_xy.so
ibgp42.flt    imcm72.flt  libde_bmp.so           libsc_du.so    libvs_asf.so   libvs_eshr.so  libvs_ich6.so  libvs_mbox.so  libvs_mwpf.so  libvs_pntg.so  libvs_rbs.so   libvs_soc.so   libvs_vw3.so   libvs_wp6.so   libvs_yim.so
ibjpg2.flt    imcmx2.flt  libde_ss.so            libsc_ex.so    libvs_bdr.so   libvs_exe2.so  libvs_ich.so   libvs_mcw.so   libvs_mwrk.so  libvs_pp12.so  libvs_rft.so   libvs_soi6.so  libvs_w12.so   libvs_wpf.so   libvs_zip.so
ibpcd2.flt    imdsf2.flt  libde_tree.so          libsc_fa.so    libvs_bmp.so   libvs_fax.so   libvs_img.so   libvs_mdb.so   libvs_one.so   libvs_pp2.so   libvs_rfx.so   libvs_soi.so   libvs_w6.so    libvs_wpg2.so  libwv_core.so
ibpsd2.flt    imfmv2.flt  libde_vect.so          libsc_fi.so    libvs_cgm.so   libvs_fcd.so   libvs_iwok.so  libvs_mif.so   libvs_ow.so    libvs_pp7.so   libvs_riff.so  libvs_sow.so   libvs_w97.so   libvs_wpg.so
ibxbm2.flt    imgdf2.flt  libde_wp.so            libsc_fmt.so   libvs_dbs.so   libvs_fcs.so   libvs_iwon.so  libvs_mime.so  libvs_pbm.so   libvs_pp97.so  libvs_rtf.so   libvs_spt.so   libvs_wbmp.so  libvs_wpl.so
ibxpm2.flt    imgem2.flt  libex_pagelayout.so    libsc_fnt.so   libvs_dez.so   libvs_fft.so   libvs_iwop.so  libvs_mm4.so   libvs_pcl.so   libvs_ppl.so   libvs_sam.so   libvs_ssml.so  libvs_wg2.so   libvs_wpml.so
ibxwd2.flt    imigs2.flt  libfreetype.so         libsc_fut.so   libvs_dif.so   libvs_flw.so   libvs_iwp.so   libvs_mmfn.so  libvs_pcx.so   libvs_pstf.so  libvs_sc5.so   libvs_swf.so   libvs_wk4.so   libvs_wpw.so
imcd32.flt    immet2.flt  libfreetype.so.6       libsc_ind.so   libvs_drw.so   libvs_fwk.so   libvs_jbg2.so  libvs_mm.so    libvs_pdfi.so  libvs_pst.so   libvs_sdw.so   libvs_taz.so   libvs_wk6.so   libvs_ws2.so
imcd42.flt    impif2.flt  libfreetype.so.6.3.13  libsc_lo.so    libvs_dxla.so  libvs_gdsf.so  libvs_jp2.so   libvs_mpp.so   libvs_pdf.so   libvs_qad.so   libvs_shw3.so  libvs_text.so  libvs_wks.so   libvs_ws.so
imcd52.flt    imps_2.flt  liboc_emul.so          libsc_ut.so    libvs_dxlm.so  libvs_gif.so   libvs_jw.so    libvs_mp.so    libvs_pdx.so   libvs_qa.so    libvs_smd.so   libvs_tga.so   libvs_wmf.so   libvs_xl12.so
imcd62.flt    impsi2.flt  libos_pdf.so           libsc_xp.so    libvs_dx.so    libvs_gzip.so  libvs_leg.so   libvs_msg.so   libvs_pfs.so   libvs_qp6.so   libvs_sms.so   libvs_tif6.so  libvs_wml.so   libvs_xl5.so
imcd72.flt    impsz2.flt  libsc_anno.so          libvs_acd2.so  libvs_emf.so   libvs_hgs.so   libvs_lwp7.so  libvs_msw.so   libvs_pgl.so   libvs_qp9.so   libvs_smt.so   libvs_tw.so    libvs_wm.so    libvs_xlsb.so
```

**Comment by lakenen** [comment]
Yep, looks like I have everything there as well. I guess at this poing I'm just going to give up and move back to the other machine. Thanks for the help! P.S. is it even possible to get Oracle support on this? I can't find anything useful on the website, but I've been having problems with images in converted pdfs having really bad aliasing issues... I've tried adding various combinations of options like [this](http://docs.oracle.com/cd/E26550_01/dev.840/e12886/e02_pxoptions.htm#sthref250) and [this](http://docs.oracle.com/cd/E26550_01/dev.840/e12886/e02_pxoptions.htm#sthref469), but still no luck.

**Comment by ryancole** [comment]
Support for these libs is basically non-existent. I believe Oracle has a forum for this software, but I've posted on it before and it didn't seem very active. Like I mentioned above, I ran into the same error while trying to make a node module for Oracle image exporter lib, and I just gave up. I honestly think Oracle left out a necessary export filter from their image exporter SDK. No way to get help, though. :(
